prow/README.md

```diff
@@ -1,41 +1,72 @@
 # Prow deployment
 
-This directory contains manifests used for the deployment of the Prow cluster.
+This directory contains the manifests used for the deployment of the Prow
+cluster.
 
-## Updating Prow
+## Upgrading Prow
 
-The core Prow components are automatically built from our own fork of the test-infra
-repository, and applied to our production build cluster.
+The Kubernetes [Prow
+deployment](https://github.com/kubernetes/test-infra/tree/master/prow) is
+automatically deployed, but all the other projects like Knative, Istio, and
+cert-manager do the deployment manually.
 
-In order to upgrade Prow to a new version, you will first need to change the
-[WORKSPACE](../WORKSPACE) file in the root of this repository to reference
-the desired revision.
+Here is the process to upgrade Prow:
 
-For example, you should change the `commit` here appropriately:
+1. ⚠️ You must be given the role `roles/container.developer` on the
+   [jetstack-build-infra](https://console.cloud.google.com/home/dashboard?project=jetstack-build-infra)
+   project. You must be able to run `kubectl` commands on the
+   [github-build-infra](https://console.cloud.google.com/kubernetes/clusters/details/europe-west1-b/github-build-infra/details?project=jetstack-build-infra)
+   cluster.
+2. Clone this repo:
 
-```
-git_repository(
-    name = "test_infra",
-    commit = "a8cee5a60a2d9476341cf843867221a8bd18a3e8",
-    remote = "https://github.com/kubernetes/test-infra.git",
-)
-```
+   ```sh
+   git clone https://github.com/jetstack/testing
+   cd testing
+   ```
 
-Once this is done, you can use Bazel to build, push and deploy the relevant new
-images:
+3. Pick a commit of Prow on
+   [kubernetes/test-infra](https://github.com/kubernetes/test-infra). That's the
+   commit to which you will be upgrading to. We use a commit instead of a git
+   tag due to Prow not having releases. For example, let's pick
+   [eca83d2ac](https://github.com/kubernetes/test-infra/commit/eca83d2ac).
+4. Review
+   [ANNOUNCEMENTS.md](https://github.com/kubernetes/test-infra/blob/master/prow/ANNOUNCEMENTS.md)
+   and look for anything that changed between the previous commit and your new
+   commit.
+5. Open a PR to [jetstack/infra](https://github.com/jetstack/infra) with the
+   update to the `commit` field in the file [WORKSPACE](../WORKSPACE). For
+   example, if you want to be upgrading from Prow
+   [a8cee5a60](https://github.com/kubernetes/test-infra/commit/a8cee5a60) to
+   [eca83d2ac](https://github.com/kubernetes/test-infra/commit/eca83d2ac), the
+   change to WORKSPACE is:
 
-```
-$ bazel run //prow/cluster:production.apply
-```
+   ```diff
+   git_repository(
+       name = "test_infra",
+   -   commit = "a8cee5a60a2d9476341cf843867221a8bd18a3e8",
+   +   commit = "eca83d2ac2b48c2732aab0c90c6eff6e564d4a21",
+       remote = "https://github.com/kubernetes/test-infra.git",
+   )
```

Comment:
This step is not necessary :) you need to change the `prow/version` file instead

Reply:
https://github.com/jetstack/testing/blob/449fa5d671c1bf2f98ce2f3f1a5cab455b93b70b/images/builder/ci-runner.sh#L78-L84 <- this is the only place that the test_infra repository dependency is actually used

Reply:
Fixed in 6ba1687! Let me know if it looks better